### PR TITLE
Add override command to modify host properties

### PR DESCRIPTION
**Pull request by hectorj2f**
This PR aims to add support to update the properties of the hosts such as its CoreOS version, yochu or docker version, etc. Nowadays, it is only possible to update those properties by modifying its configuration file. 

- [x] Change the binaries version by using TemplateEnv
```
    YOCHU_VERSION={{index .TemplatesEnv "yochu_version"}}
    DOCKER_VERSION={{index .TemplatesEnv "docker_version"}}
    ETCD_VERSION={{index .TemplatesEnv "etcd_version"}}
    FLEET_VERSION={{index .TemplatesEnv "fleet_version"}}
    RKT_VERSION={{index .TemplatesEnv "rkt_version"}}
    K8S_VERSION={{index .TemplatesEnv "k8s_version"}}
```
- [x] Add support to handle multiple CoreOS versions
- [x] Add command to change the machine state
- [x] Add etcd discovery URL property
- [x] Update the README to add these two new commands.

Related issues:

https://github.com/giantswarm/mayu/issues/16
https://github.com/giantswarm/mayu/issues/17
https://github.com/giantswarm/mayu/issues/20
https://github.com/giantswarm/mayu/pull/47

### CODE REVIEW

**Comment by hectorj2f** [comment]
Ping @hugo how can we avoid to restart mayu to force a reconfiguration? I added the functionality to change the state

**Comment by htr** [comment]
using the API to change the state should be enough.

**Comment by hectorj2f** [comment]
@htr I don't follow. we are now able to change the state via `mayuctl`. However, if I reboot a machine which is in configured state, the process of re-installation don't take place automatically.

**Comment by htr** [comment]
ok, checked the whole diff. dnsmasq has to be restarted :)

**Comment by hectorj2f** [comment]
@htr there we go ;)... so then the solution is to restart the dnsmasq. Can we have other side-effects of doing this ? Can we affect to other containers ? Or you don't see a clean way of doing it. :dancer:

**Comment by htr** [comment]
no practical side-effects. This is the way to go without replacing the whole dnsmasq.

**Comment by hectorj2f** [comment]
ping @htr

**Comment by htr** [comment]
for me the reconfigure should be implicit when a host's state changed

**Comment by hectorj2f** [comment]
@htr I agree, but from any state transition ? or just simply `unknown|running|installed` to `configured`

**Comment by hectorj2f** [comment]
@htr then I'll remove `reconfigure` bc I don't see the need for this command. veto ?

**Comment by htr** [comment]
no veto.

**Comment by hectorj2f** [comment]
@htr I addressed your comment https://github.com/giantswarm/mayu/pull/47#issuecomment-211352350

File: pxemgr/ipxe_handlers.go [hunk 1]
```diff
@@ -475,6 +479,104 @@ func (mgr *pxeManagerT) setIPMIAddr(serial string, w http.ResponseWriter, r *htt
...
+	if err != nil {
+		w.WriteHeader(500)
+		w.Write([]byte("committing updated host overrides failed"))
+		return
+	}
+	mgr.cluster.Update()
+	w.WriteHeader(202)
+}
+
+func (mgr *pxeManagerT) reconfigureHosts(w http.ResponseWriter, r *http.Request) {
```
Comment (htr) [inline pxemgr/ipxe_handlers.go]: where is `reconfigureHosts` used ?

apart from this, LGTM

Comment (hectorj2f) [inline pxemgr/ipxe_handlers.go]: u're right 👍

**Pull request merged**